src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -779,66 +818,99 @@ private BoundPattern BindVarPattern(VarPatternSyntax node, TypeSymbol inputType,
+                            // do not correctly treat the non-generic struct `System.ValueTuple` as a tuple type.  We explicitly perform the tests
+                            // required to identify it.  When that bug is fixed we should be able to remove this if statement.
+                            addSubpatternsForTuple(ImmutableArray<TypeSymbolWithAnnotations>.Empty);
+                        }
+                        else if (strippedInputType.IsTupleType)
+                        {
+                            // It is a tuple type. Work according to its elements
+                            addSubpatternsForTuple(strippedInputType.TupleElementTypes);
+                        }
                         else
                         {
                             // It is not a tuple type. Seek an appropriate Deconstruct method.
-                            var inputPlaceholder = new BoundImplicitReceiver(node, inputType); // A fake receiver expression to permit us to reuse binding logic
+                            var inputPlaceholder = new BoundImplicitReceiver(node, strippedInputType); // A fake receiver expression to permit us to reuse binding logic
+                            var deconstructDiagnostics = DiagnosticBag.GetInstance();
                             BoundExpression deconstruct = MakeDeconstructInvocationExpression(
-                                tupleDesignation.Variables.Count, inputPlaceholder, node, diagnostics, outPlaceholders: out ImmutableArray<BoundDeconstructValuePlaceholder> outPlaceholders);
+                                tupleDesignation.Variables.Count, inputPlaceholder, node, deconstructDiagnostics, outPlaceholders: out ImmutableArray<BoundDeconstructValuePlaceholder> outPlaceholders,
+                                out bool anyDeconstructCandidates);
+                            if (!anyDeconstructCandidates && ShouldUseITuple(node, strippedInputType, diagnostics, out var iTupleType, out var iTupleGetLength, out var iTupleGetItem))
+                            {
+                                // There was no applicable candidate Deconstruct, and the constraints for the use of ITuple are satisfied.
+                                // Use that and forget any errors from trying to bind Deconstruct.
+                                deconstructDiagnostics.Free();
+                                BindITupleSubpatterns(tupleDesignation, subPatterns, diagnostics);
+                                return new BoundITuplePattern(node, iTupleGetLength, iTupleGetItem, subPatterns.ToImmutableAndFree(), strippedInputType, hasErrors);
+                            }
+                            else
+                            {
+                                diagnostics.AddRangeAndFree(deconstructDiagnostics);
+                            }
+
                             deconstructMethod = deconstruct.ExpressionSymbol as MethodSymbol;
+                            if (!hasErrors)
```

Comment:
>) [](start = 42, length = 1)

braces?

Reply:
I'd rather remove unnecessary braces.

---
In reply to: [234839920](https://github.com/dotnet/roslyn/pull/31056#discussion_r234839920) [](ancestors = 234839920)